safir/src/safir/metrics/_config.py

```diff
@@ -131,7 +153,11 @@ class DisabledMetricsConfiguration(BaseMetricsConfiguration):
 
     @override
     def make_manager(
-        self, logger: BoundLogger | None = None
+        self,
+        logger: BoundLogger | None = None,
+        *,
+        kafka_broker: KafkaBroker | None = None,
+        kafka_admin_client: AIOKafkaAdminClient | None = None,
```

Comment:
You could do something like:
```python
@dataclass
class KafkaClients:
    broker: KafkaBroker
    admin_client: AIOKafkaAdminClient

def make_manager(self, logger: str, kafka_clients: KafkaClients | None):
    etc
```
Then the "both or neither" condition would be verified at the type level... at the expense of a weird interface. I don't have a strong preference either way!

Reply:
Oh, yeah, that's better. I was trying to figure out a way to do that and that one didn't occur to me. Thanks!